packages/editor-ui/src/views/canvasHelpers.ts

```diff
@@ -511,20 +499,32 @@ export const getOutputSummary = (data: ITaskData[], nodeConnections: NodeInputCo
 		}
 
 		run.data.main.forEach((output: INodeExecutionData[] | null, i: number) => {
-			if (!nodeConnections[i]) {
+			const sourceOutputIndex = i;
+
+			if (!outputMap[sourceOutputIndex]) {
+				outputMap[sourceOutputIndex] = {};
+			}
+
+			if (!outputMap[sourceOutputIndex][NODE_OUTPUT_DEFAULT_KEY]) {
+				outputMap[sourceOutputIndex][NODE_OUTPUT_DEFAULT_KEY] = {};
+				outputMap[sourceOutputIndex][NODE_OUTPUT_DEFAULT_KEY][0] = {
+					total: 0,
+					iterations: 0,
+				};
+			}
+
+			outputMap[sourceOutputIndex][NODE_OUTPUT_DEFAULT_KEY][0].total += output ? output.length : 0;
+			outputMap[sourceOutputIndex][NODE_OUTPUT_DEFAULT_KEY][0].iterations += output ? 1 : 0;
```

Comment:
maybe assigning `outputMap[sourceOutputIndex][NODE_OUTPUT_DEFAULT_KEY][0]` to a temporary variable for readability

Reply:
updated